### PR TITLE
powerline-go: add zsh integration

### PR DESCRIPTION
**Pull request by tinybeachthor**
### Description

Add zsh integration for powerline-go.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

### CODE REVIEW

File: modules/programs/powerline-go.nix [hunk 1]
```diff
@@ -106,17 +106,38 @@ in {
...
+      mkIf (cfg.enable && config.programs.bash.enable) ''
+        function _update_ps1() {
+          local old_exit_status=$?
+          PS1="$(${pkgs.powerline-go}/bin/powerline-go -error $old_exit_status ${commandLineArguments})"
+          ${cfg.extraUpdatePS1}
+          return $old_exit_status
+        }
+
+        if [ "$TERM" != "linux" ]; then
+          PROMPT_COMMAND="_update_ps1;$PROMPT_COMMAND"
```
Comment (berbiche) [inline modules/programs/powerline-go.nix:120]: Can the prompt command end with a trailing semicolon?

If not:

```suggestion
          PROMPT_COMMAND="_update_ps1\${PROMPT_COMMAND:+';'}$PROMPT_COMMAND"
```

**Review by berbiche — APPROVED** [review]
LGTM!

Thanks for the contribution

File: modules/programs/powerline-go.nix (hunk 1 above)
Comment (tinybeachthor) [reply modules/programs/powerline-go.nix:120]: That code is straight from the [powerline-go README](https://github.com/justjanne/powerline-go).

Tested it to be sure, and the trailing semicolon is no problem.

**Comment by rycee** [comment]
Thanks! Merged to master.

**Pull request merged**